api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -96,7 +96,7 @@ public AppFrameworkServiceImpl(AllAppTemplates allAppTemplates, AllAppDescriptor
 		this.allUserApps = allUserApps;
 
 		if (javascriptEngine == null) {
-			javascriptEngine = new jdk.nashorn.api.scripting.NashornScriptEngineFactory().getScriptEngine();
+			javascriptEngine = new ScriptEngineManager().getEngineByName("graal.js");
```

Comment:
Did you confirm that the above does not return NULL on Java 17?

Reply:
When you use the correct dependency, we can remove this block because `javascriptEngine` will never be NULL